src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -95,33 +115,43 @@ export const selectRedelegationCryptoAmountByDenom = createSelector(
 export const selectUnbondingEntriesByAccountSpecifier = createDeepEqualOutputSelector(
   selectStakingDataByAccountSpecifier,
   selectValidatorAddress,
-  (
-    stakingData,
-    validatorAddress
-  ): Array<{ denom: string; amount: string; completionTime: number }> => {
+  (stakingData, validatorAddress): chainAdapters.cosmos.UndelegationEntry[] => {
     if (!stakingData || !stakingData.undelegations) return []
 
     return (
-      stakingData.undelegations
-        .find(({ validator }) => validator.address === validatorAddress)
-        ?.entries.map(undelegationEntry => ({
-          denom: ASSET_ID_TO_DENOM[undelegationEntry.assetId],
-          amount: undelegationEntry.amount,
-          completionTime: undelegationEntry.completionTime
-        })) || []
+      stakingData.undelegations.find(({ validator }) => validator.address === validatorAddress)
+        ?.entries || []
     )
   }
 )
 
-export const selectUnbondingCryptoAmountByDenom = createSelector(
+export const selectAllUnbondingsEntriesByAssetId = createDeepEqualOutputSelector(
+  selectStakingDataByAccountSpecifier,
+  selectAssetId,
+  (stakingData, selectedAssetId): Record<string, chainAdapters.cosmos.UndelegationEntry[]> => {
+    if (!stakingData || !stakingData.undelegations) return {}
+
+    return stakingData.undelegations.reduce((acc, { validator, entries }) => {
+      if (!acc[validator.address]) {
+        acc[validator.address] = []
+      }
+
+      acc[validator.address].push(...entries.filter(x => x.assetId === selectedAssetId))
+
+      return acc
+    }, {} as Record<string, chainAdapters.cosmos.UndelegationEntry[]>)
```

Comment:
either define `const initial: Record<strin....` and feed it in or do `reduce<Record<string...>` - don't cast here

Reply:
done